File: listings/chapter08/euclid_gcd.py
def gcd(m, n):
    while n != 0:
        print(f"{m}, {n}")
        m, n = n, m % n
    return m

if __name__ == '__main__':
    while True:
        m_input = input("Erste Zahl: ")
        n_input = input("Zweite Zahl: ")
        try:
            m = int(m_input)
            n = int(n_input)
        except ValueError:
            print("Bitte nur Zahlen eingeben!")
            continue
        print(f"GGT von {m} und {n}: {gcd(m, n)}")
        again = input("Noch mal (j/n)? ")
        if again == "n" or again == "N":
            break

File: listings/chapter08/EuclidGCD.java
public class EuclidGCD {
    public static int gcd(int m, int n) {
        while (n != 0) {
            int temp = n;
            n = m % n;
            m = temp;
        }
        return m;
    }

    public static void main(String[] args) {
        if (args.length < 2) {
             System.out.println("Verwendung: java EuclidGCD m n");
             System.exit(1);
        }
        int m = 0;
        int n = 0;
        try {
            m = Integer.parseInt(args[0]);
            n = Integer.parseInt(args[1]);
        } catch(NumberFormatException e) {
            System.out.println("Bitte nur Zahlen angeben");
            System.exit(2);
        }
        int gcd = gcd(m, n);
        System.out.printf("GGT von %d und %d: %d\n", m, n, gcd);
    }
}
